### PR TITLE
Added QuickSort [Java]

### LINKED ISSUE #82: QuickSort [java]
https://en.wikipedia.org/wiki/Quicksort

### PR DESCRIPTION
**Pull request by jsroyal**
Fixed  #82

#### By submitting this pull request I confirm I've read and complied with the below requirements.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

### CODE REVIEW

File: quicksort/QuickSort.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+public class QuickSort {
+
+    static int partition(int[] a, int first, int last) {
+        int pivot = a[last];
+        while(first<last) {
```
Comment (Monal5031) [inline quicksort/QuickSort.java]: Please Add a space between bracket and `while` so as to make it look different from function's (method's). Also add space before and after any binary operator

File: quicksort/QuickSort.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+public class QuickSort {
+
+    static int partition(int[] a, int l, int r) {
```
Comment (aviaryan) [inline quicksort/QuickSort.java]: @jsroyal This means that specify inheritance level `public` or `private` before the `partition` method.

File: quicksort/QuickSort.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
+public class QuickSort {
```
Comment (aviaryan) [inline quicksort/QuickSort.java]: @jsroyal What about adding a `package main` at the top. This will fix this issue.

Comment (jsroyal) [inline quicksort/QuickSort.java]: >package main
Why it need to write? can you explain it?

Comment (aviaryan) [inline quicksort/QuickSort.java]: That would have fixed this error. But doesn't matter, now I will remove this check from codacy.

File: quicksort/QuickSort.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            if (l <= r) {
+                int temp = a[l];
+                a[l] = a[r];
+                a[r] = temp;
+            }
+        }
+        return l; //pivot index
+    }
+
+    static void quickSort(int[] a, int l, int r) {
```
Comment (aviaryan) [inline quicksort/QuickSort.java]: Make it  `public static void`

**Comment by jsroyal** [comment]
@aviaryan Could please remove adding package test case.
What is second one?

**Comment by aviaryan** [comment]
Brilliant work, @jsroyal !!

**Pull request merged**